src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -92,33 +89,39 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
         };
 
         public Server() {
-            _queue = new AnalysisQueue();
-            _queue.UnhandledException += Analysis_UnhandledException;
+            AnalysisQueue = new AnalysisQueue();
+            AnalysisQueue.UnhandledException += Analysis_UnhandledException;
+
             _pendingAnalysisEnqueue = new VolatileCounter();
-            _parseQueue = new ParseQueue();
+            ParseQueue = new ParseQueue();
             _pendingParse = new Dictionary<IDocument, VolatileCounter>();
             _editorFiles = new EditorFiles(this);
             _extensions = new ConcurrentDictionary<string, ILanguageServerExtension>();
+
+            _disposableBag
+                .Add(() => ProjectFiles.Dispose())
```

Comment:
`DisposableBag` has an overload of `Add` that accepts `IDisposable`, so `_disposableBag.Add(ProjectFiles)` is enough.